arwen/contexts/managedType.go

```diff
@@ -56,6 +58,19 @@ func NewManagedTypesContext(host arwen.VMHost) (*managedTypesContext, error) {
 	return context, nil
 }
 
+func initRandomizer(host arwen.VMHost) *math.SeedRandReader {
+	randomizer, _ := math.NewSeedRandReader(append(host.Blockchain().CurrentRandomSeed(), host.Runtime().GetCurrentTxHash()...))
+
+	return randomizer
+}
+
+func (context *managedTypesContext) GetRandReader(ctx unsafe.Pointer) *math.SeedRandReader {
+	if context.Reader == nil {
```

Comment:
do check.IfNil - instead of simple nil check here. make reader implement IsInterfaceNil. Make Reader private as well.